avocado/core/nrunner.py

```diff
@@ -576,20 +575,47 @@ def unittest_name(self):
             if module.startswith(os.path.sep):
                 module = module[1:]
         module = module.replace(os.path.sep, ".")
-        return '%s.%s' % (module, class_method)
+        klass, method = class_method.rsplit('.', maxsplit=1)
```

Comment:
I also put a probe here, and found that the same command line above hit this line of code 6 times... it seems a bit excessive.  I've checked the previous version and `_uri_to_unittest_name()` is called just once.  I understand that this is a trade off when using properties, but again, seems a bit excessive.  Maybe there's a low hanging fruit optimization there.

Reply:
`_uri_to_unittest_name()` as replaced by `unittest_name()` that is also called one once. The problem is that now I'm creating a standard in terms of "source of truth" in other parts of the code, to avoid changes in multiple places. Anyway, I have changed a little bit the code and reduced by half the number of calls to `unittest()` property.